Applications/TextEditor/TextEditorWidget.cpp

```diff
@@ -121,5 +143,35 @@ void TextEditorWidget::open_sesame(const String& path)
 
     window()->set_title(String::format("Text Editor: %s", path.characters()));
     m_editor->set_text(String::copy(file.read_all()));
+    m_path = path;
 }
 
+bool TextEditorWidget::file_exists(const StringView& path)
+{
+    struct stat st;
+    int rc = stat(path.characters_without_null_termination(), &st);
+    if (rc < 0) {
+        if (errno == ENOENT)
+            return false;
+    }
+    if (rc == 0) {
+        return true;
+    }
+    return false;
+}
+
+bool TextEditorWidget::create_file(const StringView& path)
+{
+    int fd = open_with_path_length(path.characters_without_null_termination(), path.length(), O_CREAT, 0100644);
```

Comment:
This should be done differently and the logic should be in `GFilePicker` but here's a tip anyways:
You can pass `O_CREAT | O_EXCL` together. `O_EXCL` will cause the syscall to fail if the file already exists.